### PR TITLE
Changes around language keys

### PR DESCRIPTION
**Pull request by SimonSimCity**
Here are some changes I would add to #310:

- [Added types for language keys](https://github.com/bcc-code/bmm-web/commit/f1ea299c081379997221b7b81d71222cf4596fe1)
- [Remove unneeded type identified by types](https://github.com/bcc-code/bmm-web/commit/e73d5711deadd626a647021de9d4197b19cb9aba)
- [Integrated language-dict into getLocalizedLanguageName](https://github.com/bcc-code/bmm-web/commit/d0bdffdec83a59a765b841d46d6bf70ab1065172)
- [Remove unnecessary type association](https://github.com/bcc-code/bmm-web/commit/93f1808fc7aae0ee577d2b9978ec67a58c601a4b)
- [Move initialization of the language to a central place](https://github.com/bcc-code/bmm-web/commit/59d2e02e9d19e31aaf89f0b1fbd95c1efc916a3c)

### CODE REVIEW

File: components/ProfileMenu.vue [hunk 1]
```diff
@@ -261,15 +255,13 @@ const closeContentLanguageDialog = () => {
...
+            <select v-model="profileStore.uiLanguage" class="py-1">
               <option
                 v-for="(lang, i) in $i18n.availableLocales"
                 :key="`Lang${i}`"
                 :value="lang"
               >
-                {{
-                  `${languageDictionary[lang]?.NativeName} (${languageDictionary[lang]?.EnglishName})`
-                }}
+                {{ getLocalizedLanguageName(lang) }}
```
Comment (kkuepper) [inline components/ProfileMenu.vue:264]: We should still show the native name in brackets

File: components/ProfileMenu.vue [hunk 2]
```diff
@@ -314,15 +306,16 @@ const closeContentLanguageDialog = () => {
...
+                    (x) => !contentLanguages.includes(x),
+                  ),
+                ]"
+                :key="`Lang${j}`"
                 :value="lang"
               >
-                {{
-                  `${languageDictionary[lang]?.NativeName} (${getLocalizedLanguageName(lang)})`
-                }}
+                {{ getLocalizedLanguageName(lang) }}
```
Comment (kkuepper) [inline components/ProfileMenu.vue:318]: also show native name

**Review by kkuepper — CHANGES_REQUESTED** [review]

File: utils/getLocalizedLanguageName.ts [hunk 3]
```diff
@@ -1,8 +1,50 @@
+import type availableContentLanguages from "./availableContentLanguages";
+
+const englishLanguageNames: {
```
[the remark this answers is not in the document]
Comment (SimonSimCity) [reply utils/getLocalizedLanguageName.ts:3]: Should the alternative then be `<english> (<native>)`, or how should it be?

File: components/ProfileMenu.vue (hunk 1 above)
Comment (SimonSimCity) [reply components/ProfileMenu.vue:264]: Always, or just if it's not translated?

File: components/ProfileMenu.vue (hunk 2 above)
Comment (SimonSimCity) [reply components/ProfileMenu.vue:318]: Always, or just if it's not translated?

File: components/ProfileMenu.vue (hunk 1 above)
Comment (kkuepper) [reply components/ProfileMenu.vue:264]: Always. Imagine that you selected Russian. Then you have no idea which of the 25 listed languages is english or german. Therefore it makes a lot of sense to show the native name as well, when selecting a language.

Comment (SimonSimCity) [reply components/ProfileMenu.vue:264]: Should we unify this approach, or where should it differ? Should it be only different for showing the ui language?

Approach: For this I'd extend the function `getLocalizedLanguageName()` to take a second argument which is the language. If not provided, it takes the default, which is the one selected as ui language.

File: utils/getLocalizedLanguageName.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkuepper) [reply utils/getLocalizedLanguageName.ts:3]: I think getLocalizedLanguageName should should only return <translated> or <english>.
The UI has to display it as <translated> (<native>).

Screenshot 2024-02-27 at 8 12 49 AM
On this view we don't want to show the native name in brackets. (Also see the design in Figma)

File: components/ProfileMenu.vue (hunk 1 above)
Comment (kkuepper) [reply components/ProfileMenu.vue:264]: In Figma, native name in brackets is used for UI language and content language.

File: utils/getLocalizedLanguageName.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (SimonSimCity) [reply utils/getLocalizedLanguageName.ts:3]: Can I interpret it as "native", if a system can translate the language to itself? 😎 Then my approach above would be valid.

File: utils/getLocalizedLanguageName.ts [hunk 4]
```diff
@@ -1,8 +1,53 @@
...
+};
+
+function getEnglishLanguageName(k: string) {
+  const languageNames: { [key: string]: string } = englishLanguageNames;
+  return languageNames[k];
+}
+
+export default function getLocalizedLanguageName(
+  key: MaybeRef<string>,
+  useNativeName?: boolean,
```
Comment (kkuepper) [inline utils/getLocalizedLanguageName.ts:43]: I suggest using a boolean parameter. I think that's more readable

File: components/track/TrackChangeLocale.vue
Comment (kkuepper) [inline components/track/TrackChangeLocale.vue:1]: this file has a merge conflict

File: utils/getLocalizedLanguageName.ts (hunk 4 above)
Comment (SimonSimCity) [reply utils/getLocalizedLanguageName.ts:43]: Isn't this a boolean parameter..?

Comment (SimonSimCity) [reply utils/getLocalizedLanguageName.ts:43]: I guess you've already implemented your suggestion...

**Review by kkuepper — APPROVED** [review]

**Pull request merged**